lib/misc/cells.js

```diff
@@ -0,0 +1,64 @@
+(function() {
+  var scopes, values;
+
+  scopes = require('./scopes');
+
+  values = require('underscore-plus').values;
+
+  module.exports = {
+    getRange: function(ed) {
+      var buffer, cursor, end, regex, regexString, start;
+      buffer = ed.getBuffer();
+      start = buffer.getFirstPosition();
+      end = buffer.getEndPosition();
+      regexString = atom.config.get('julia-client.cellDelimiter');
+      regex = new RegExp(regexString);
+      cursor = ed.getCursorBufferPosition();
+      if (cursor.row > 0) {
+        buffer.backwardsScanInRange(regex, [start, cursor], function(arg) {
```

Comment:
You also shouldn't include the cell delimiter line in the range here -- if you do, then previous results will disappear since the cell ranges overlap.